pytest_flakefinder.py

```diff
@@ -50,8 +50,16 @@ def __init__(self, config):
     def pytest_generate_tests(self, metafunc):
         """For all true pytest tests use metafunc to add all the duplicates."""
         # This is safer because otherwise test with fixtures might not be setup correctly.
-        for _ in range(self.flake_runs):
-            metafunc.addcall()
+        # Parameterization requires the test function to accept the permutation as
+        # either an argument or depend on it as a fixture. Use fixture so the test
+        # function signature is not changed. Suffix with timestamp to reduce odds of
+        # collision with other fixture names.
+        fixture_name = 'flakefinder-{}'.format(time.time())
```

Comment:
What is the purpose of time.time() here? To prevent collision with an actual fixture named "flakefinder"? Would prefer if we just used something like metafunc.function.__name__ to append and disambiguate. Can we use an underscore instead of a dash so it's a valid python identifier?

Looking at the documentation, I don't see any good example of appending to fixturenames. I'd be worried that this usage (though it works today) might be an abuse that backfires in the future as the pytest dev team changes the library.

Reply:
This risk is probably worth it, since they don't really document any other way to pull this off.

Reply:
Yes, to avoid potential fixture name collisions. How about this:
`'__flakefinder_{funcname}'.format(metafunc.function.name)`?